### ISSUE TITLE
The input line is too long. The syntax of the command is incorrect.

### ISSUE THREAD

**Issue by bmatthew1** [NateIsStalling/Basic4GLj#37]
Running the latest build on the current version of Windows.

When running `Basic4GLj.bat` from a Command-Prompt, I get the following error...

```
The input line is too long.
The syntax of the command is incorrect.
```

In `Basic4GLj.bat` the following line contains too many characters...

```
set CLASSPATH=%APP_HOME%\lib\app.jar;%APP_HOME%\lib\RSyntaxTextArea.jar;%APP_HOME%\lib\library-1.0-SNAPSHOT.jar;%APP_HOME%\lib\lwjgl-javadoc.jar;%APP_HOME%\lib\lwjgl-natives-macos-arm64.jar;%APP_HOME%\lib\lwjgl-openal-javadoc.jar;%APP_HOME%\lib\lwjgl-natives-windows-x86.jar;%APP_HOME%\lib\lwjgl-stb-natives-windows-arm64.jar;%APP_HOME%\lib\lwjgl-glfw-natives-macos-arm64.jar;%APP_HOME%\lib\lwjgl-glfw-natives-windows-x86.jar;%APP_HOME%\lib\lwjgl-openal-natives-linux-arm64.jar;%APP_HOME%\lib\lwjgl-openal-natives-macos.jar;%APP_HOME%\lib\lwjgl-opengl-javadoc.jar;%APP_HOME%\lib\lwjgl-assimp-natives-macos-arm64.jar;%APP_HOME%\lib\lwjgl-assimp-natives-windows-x86.jar;%APP_HOME%\lib\lwjgl-glfw-natives-linux.jar;%APP_HOME%\lib\lwjgl-opengl-natives-windows.jar;%APP_HOME%\lib\lwjgl-stb-natives-windows.jar;%APP_HOME%\lib\lwjgl-openal-natives-windows-arm64.jar;%APP_HOME%\lib\lwjgl-stb.jar;%APP_HOME%\lib\lwjgl-stb-natives-linux.jar;%APP_HOME%\lib\lwjgl-glfw-natives-windows.jar;%APP_HOME%\lib\lwjgl-natives-linux-arm64.jar;%APP_HOME%\lib\lwjgl-glfw.jar;%APP_HOME%\lib\lwjgl-opengl-natives-linux-arm32.jar;%APP_HOME%\lib\lwjgl-stb-natives-linux-arm32.jar;%APP_HOME%\lib\lwjgl-assimp-natives-linux.jar;%APP_HOME%\lib\lwjgl-natives-macos.jar;%APP_HOME%\lib\lwjgl-openal-natives-macos-arm64.jar;%APP_HOME%\lib\lwjgl-glfw-natives-linux-arm64.jar;%APP_HOME%\lib\lwjgl-openal-natives-windows-x86.jar;%APP_HOME%\lib\lwjgl-assimp-javadoc.jar;%APP_HOME%\lib\lwjgl-glfw-javadoc.jar;%APP_HOME%\lib\lwjgl.jar;%APP_HOME%\lib\CodecIBXM.jar;%APP_HOME%\lib\lwjgl-assimp-natives-linux-arm64.jar;%APP_HOME%\lib\lwjgl-opengl-natives-macos.jar;%APP_HOME%\lib\lwjgl-opengl-natives-windows-arm64.jar;%APP_HOME%\lib\lwjgl-assimp-natives-windows-arm64.jar;%APP_HOME%\lib\lwjgl-opengl-natives-macos-arm64.jar;%APP_HOME%\lib\lwjgl-opengl-natives-windows-x86.jar;%APP_HOME%\lib\CodecWav.jar;%APP_HOME%\lib\lwjgl-assimp-natives-macos.jar;%APP_HOME%\lib\lwjgl-openal-natives-linux-arm32.jar;%APP_HOME%\lib\lwjgl-natives-linux.jar;%APP_HOME%\lib\lwjgl-stb-natives-windows-x86.jar;%APP_HOME%\lib\lwjgl-stb-natives-macos-arm64.jar;%APP_HOME%\lib\lwjgl-stb-javadoc.jar;%APP_HOME%\lib\lwjgl-assimp.jar;%APP_HOME%\lib\lwjgl-glfw-natives-windows-arm64.jar;%APP_HOME%\lib\lwjgl-opengl-natives-linux.jar;%APP_HOME%\lib\lwjgl-openal.jar;%APP_HOME%\lib\CodecJOrbis.jar;%APP_HOME%\lib\lwjgl-natives-linux-arm32.jar;%APP_HOME%\lib\lwjgl-opengl-natives-linux-arm64.jar;%APP_HOME%\lib\lwjgl-natives-windows.jar;%APP_HOME%\lib\lwjgl-stb-natives-linux-arm64.jar;%APP_HOME%\lib\lwjgl-openal-natives-linux.jar;%APP_HOME%\lib\lwjgl-glfw-natives-linux-arm32.jar;%APP_HOME%\lib\lwjgl-openal-natives-windows.jar;%APP_HOME%\lib\SoundSystem.jar;%APP_HOME%\lib\lwjgl-assimp-natives-windows.jar;%APP_HOME%\lib\lwjgl-opengl.jar;%APP_HOME%\lib\lwjgl-assimp-natives-linux-arm32.jar;%APP_HOME%\lib\lwjgl-glfw-natives-macos.jar;%APP_HOME%\lib\lwjgl-natives-windows-arm64.jar;%APP_HOME%\lib\lwjgl-stb-natives-macos.jar;%APP_HOME%\lib\debugServer-1.0-SNAPSHOT.jar;%APP_HOME%\lib\compiler.jar;%APP_HOME%\lib\runtime-1.0-SNAPSHOT.jar;%APP_HOME%\lib\debugProtocol-1.0-SNAPSHOT.jar;%APP_HOME%\lib\flatlaf-extras-3.0.jar;%APP_HOME%\lib\flatlaf-3.0.jar;%APP_HOME%\lib\gson-2.10.jar;%APP_HOME%\lib\javax-websocket-server-impl-9.4.49.v20220914.jar;%APP_HOME%\lib\javax.websocket-api-1.0.jar;%APP_HOME%\lib\javax-websocket-client-impl-9.4.49.v20220914.jar;%APP_HOME%\lib\svgSalamander-1.1.3.jar;%APP_HOME%\lib\websocket-server-9.4.49.v20220914.jar;%APP_HOME%\lib\websocket-client-9.4.49.v20220914.jar;%APP_HOME%\lib\javax.websocket-client-api-1.0.jar;%APP_HOME%\lib\jetty-annotations-9.4.49.v20220914.jar;%APP_HOME%\lib\jetty-plus-9.4.49.v20220914.jar;%APP_HOME%\lib\jetty-webapp-9.4.49.v20220914.jar;%APP_HOME%\lib\jetty-servlet-9.4.49.v20220914.jar;%APP_HOME%\lib\jetty-security-9.4.49.v20220914.jar;%APP_HOME%\lib\jetty-server-9.4.49.v20220914.jar;%APP_HOME%\lib\jetty-client-9.4.49.v20220914.jar;%APP_HOME%\lib\websocket-common-9.4.49.v20220914.jar;%APP_HOME%\lib\jetty-http-9.4.49.v20220914.jar;%APP_HOME%\lib\jetty-io-9.4.49.v20220914.jar;%APP_HOME%\lib\jetty-util-ajax-9.4.49.v20220914.jar;%APP_HOME%\lib\jetty-xml-9.4.49.v20220914.jar;%APP_HOME%\lib\jetty-jndi-9.4.49.v20220914.jar;%APP_HOME%\lib\jetty-util-9.4.49.v20220914.jar;%APP_HOME%\lib\websocket-servlet-9.4.49.v20220914.jar;%APP_HOME%\lib\javax.servlet-api-3.1.0.jar;%APP_HOME%\lib\websocket-api-9.4.49.v20220914.jar;%APP_HOME%\lib\javax.annotation-api-1.3.2.jar;%APP_HOME%\lib\asm-commons-9.3.jar;%APP_HOME%\lib\asm-analysis-9.3.jar;%APP_HOME%\lib\asm-tree-9.3.jar;%APP_HOME%\lib\asm-9.3.jar
```

I'm assuming all those files on the Classpath are located in the lib folder so a Wildcard can be used and I can then get the IDE to appear on the screen...

```
set CLASSPATH=%APP_HOME%\lib\*
```
![Basic4GLj IDE running on Windows](https://github.com/NateIsStalling/Basic4GLj/assets/1728497/79250bd5-2257-47b3-a2a0-ea3a9fc09bc3)

**Comment by NateIsStalling** [comment]
Thanks for the detailed issue report @bmatthew1! 
This appears related to a known issue with the .bat scripts Gradle generates for distributions, see https://github.com/gradle/gradle/issues/1989.

Replacing the path with a wildcard appears to be a recommended workaround and shouldn't cause any issues running the app, but I'll look more into if there's an officially supported option or at least include the wildcard workaround in future releases.

**Comment by NateIsStalling** [comment]
Hi @bmatthew1, this issue should be resolved in the v0.5.0 release with commit 96466a071788f16b213b0a0f61043b03c6b9fbce, but the latest release requires Java 17 to run the Jars from the Command-Prompt. 

Installer binaries included with the v0.5.0 release are bundled with OpenJDK 17 and should avoid issues with `.bat` files as well.
